src/adapter/templates/index.ts

```diff
@@ -0,0 +1,102 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+
+import * as ts from 'typescript';
+
+/**
+ * Creates a template for the given function that replaces its arguments
+ * and generates a string to be executed where it takes expressions to be
+ * interpolated in place of arguments.  It assumes there's no shadowing
+ * going on and that the template does not reference things outside its scope.
+ *
+ * This is not pretty, but presented as an alternative to writing a bunch of
+ * raw JavaScript functions in strings.
+ *
+ * Example:
+ *
+ * ```js
+ * const multiply = (a, b) => {
+ *   return a * b;
+ * };
+ * const template = templateFunction(multiply);
+ * console.log(multiple('42', 'foo()));
+ * ```
+ *
+ * Outputs:
+ *
+ * ```
+ * (() => {
+ *   let __arg0 = 42;
+ *   let __arg1 = foo();
```

Comment:
I don't think you have to inline the arguments, I think it's possible to pass a function and then an array of arguments to call it with in that cdp API.

Reply:
One big drawback for this particular scenario there is that callFunction doesn't have access to the [command line API](https://github.com/GoogleChrome/devtools-docs/blob/master/docs/commandline-api.md), which in the REPL would be useful to have.

However, callFunction should be useful for most other scenarios :) I'll push up the (much simpler wrapper) I just wrote for making callFunction's so that we can use it later on.